### PR TITLE
Fix GetInteriorVehicleDataCapabilities parameters validation

### PR DESCRIPTION
**Pull request by AByzhynar**
- Fixed parameters validation in HMI response
- Fixed parameters validation read from default caps file
- All validation proceeded via SmartObject schema validation

### CODE REVIEW

**Review by AKalinich-Luxoft — APPROVED** [review]

File: src/components/can_cooperation/src/commands/get_interior_vehicle_data_capabilities_request.cc [hunk 1]
```diff
@@ -99,85 +99,80 @@ void GetInteriorVehicleDataCapabiliesRequest::OnEvent(
...
-      trying to read from file");
-  Json::Value request;
-  Json::Reader reader;
-  if (!reader.parse(message_->json_message(), request)) {
-    LOG4CXX_ERROR(logger_, "Failed to read capabilities from file also");
+bool GetInteriorVehicleDataCapabiliesRequest::ReadCapabilitiesFromFile(
+    const can_event_engine::Event<application_manager::MessagePtr, std::string>&
+        event) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  LOG4CXX_INFO(logger_, "Failed to get correct response from HMI!");
```
Comment (LuxoftAKutsan) [inline src/components/can_cooperation/src/commands/get_interior_vehicle_data_capabilities_request.cc]: @AByzhynar it is not a place for this  debug.

File: src/components/can_cooperation/src/commands/get_interior_vehicle_data_capabilities_request.cc [hunk 2]
```diff
@@ -99,85 +99,80 @@ void GetInteriorVehicleDataCapabiliesRequest::OnEvent(
...
+
+  VehicleCapabilities file_caps;
+  LOG4CXX_DEBUG(logger_,
+                "Read vehicle capabilities from file: "
+                    << file_caps.default_capabilities_path());
+  Json::Value interior_vehicle_data_caps_content;
```
Comment (LuxoftAKutsan) [inline src/components/can_cooperation/src/commands/get_interior_vehicle_data_capabilities_request.cc]: @AByzhynar please assign variable within declaration

File: src/components/can_cooperation/src/commands/get_interior_vehicle_data_capabilities_request.cc [hunk 3]
```diff
@@ -99,85 +99,80 @@ void GetInteriorVehicleDataCapabiliesRequest::OnEvent(
...
+  interior_vehicle_data_caps_content = file_caps.capabilities();
+  if (!interior_vehicle_data_caps_content.isObject()) {
```
Comment (LuxoftAKutsan) [inline src/components/can_cooperation/src/commands/get_interior_vehicle_data_capabilities_request.cc]: @AByzhynar 2 steps of validation is not rely readable

**Review by LuxoftAKutsan — APPROVED** [review]

File: src/components/can_cooperation/src/commands/get_interior_vehicle_data_capabilities_request.cc (hunk 3 above)
Comment (AByzhynar) [reply src/components/can_cooperation/src/commands/get_interior_vehicle_data_capabilities_request.cc]: @LuxoftAKutsan Removed

File: src/components/can_cooperation/src/commands/get_interior_vehicle_data_capabilities_request.cc (hunk 2 above)
Comment (AByzhynar) [reply src/components/can_cooperation/src/commands/get_interior_vehicle_data_capabilities_request.cc]: Done

File: src/components/can_cooperation/src/commands/get_interior_vehicle_data_capabilities_request.cc (hunk 1 above)
Comment (AByzhynar) [reply src/components/can_cooperation/src/commands/get_interior_vehicle_data_capabilities_request.cc]: @LuxoftAKutsan Moved

**Review by LuxoftAKutsan — APPROVED** [review]

**Pull request merged**